### PR TITLE
Checkers branch

### PR DESCRIPTION
**Pull request by TomerMizrachi**
we should complete all riddles

### CODE REVIEW

File: checkers/gameLogic.ts [hunk 1]
```diff
@@ -948,3 +993,18 @@ export function getInitialBoard(): Board {
     ['WM', '--', 'WM', '--', 'WM', '--', 'WM', '--'],
   ];
 }
+// function isPosOnHintLine(row: number, col: number, hint: RiddleData) {
```
Comment (fadiatamny) [inline checkers/gameLogic.ts:996]: I would remove unnecessary comments at the time of a PR

File: common/debugging.ts [hunk 2]
```diff
@@ -16,5 +16,5 @@ export const DEBUGGING_OPTIONS = {
   // (instead of waiting for 10-30 seconds)
   SHOW_HINT_AFTER_ONE_SECOND: true,
 
-  IGNORE_ASYNC_STORAGE: false,
+  IGNORE_ASYNC_STORAGE: true,
```
Comment (fadiatamny) [inline common/debugging.ts]: should return the debugging flags you used to its original value as it can cause conflicts further on.

**Review by fadiatamny — CHANGES_REQUESTED** [review]
why was there a change in the yarn-lock? you can return it to its original state so as to not cause issues with conflicts further ahead.

furthermore please use more meaningful names for your commits it's hard to understand which commit changed and added which features.

otherwise LGTM. make sure to check with  yoav as if to do the changes i recommended or not.

File: checkers/components/Game.tsx [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+} from '../gameLogic';
+import {getPossibleMoves, getStateScoreForIndex0} from '../aiService';
+import {riddleLevels} from '../riddles';
+
+let allPossibleMoves: BoardDelta[] = [];
+let selectedMovingPiece: BoardDelta;
+let pieceIsSelected = false;
+
+const styles = StyleSheet.create({
+  // See how to have a square component using aspectRatio=1
```
Comment (barakdaniel) [inline checkers/components/Game.tsx]: You can delete this comment

**Review by barakdaniel — COMMENTED** [review]
Overall the code seems great, I wouldn't change much except a few commenting for functions or variables.

File: checkers/gameLogic.ts (hunk 1 above)
Comment (TomerMizrachi) [reply checkers/gameLogic.ts:996]: it is TODO for the next pull request, we have to finish the work on the riddles, so in the next PR we will fix it.
thanks

File: common/debugging.ts (hunk 2 above)
Comment (TomerMizrachi) [reply common/debugging.ts]: will do

**Pull request closed without merging**